bundles/CoreBundle/src/DependencyInjection/Configuration.php

```diff
@@ -956,6 +957,34 @@ private function addImplementationLoaderNode(ArrayNodeDefinition $node, string $
             ->end();
     }
 
+    /**
+     * Add implementation node config with array (map[class,mapping], prefixes)
+     */
+    private function addImplementationLoaderNodeFromArrayDefinition(ArrayNodeDefinition $node, string $name): void
+    {
+        $node
+            ->children()
+                ->arrayNode($name)
+                    ->addDefaultsIfNotSet()
+                    ->children()
+                        ->arrayNode('map')
+                            ->arrayPrototype()
+                            ->children()
+                                ->scalarNode('class')->end()
+                                ->arrayNode('mapping')
```

Comment:
Wouldn't it be better to have a service that can resolves the type?

Reply:
What do you mean exactly?